### PR TITLE
Add test requirements file and remove duplicate test function

### PR DESCRIPTION
**Pull request by ahmedivy**
- Create requirements-test.txt file with torch and pytest.
- Remove duplicate `test_setitem` function in test file.

### CODE REVIEW

File: .gitignore
Comment (nizhib) [inline .gitignore:1]: The files in the repo should provide some added value, e.g. `.gitignore` could be seen as a representation of temporary files that are actually created when the code is executed as intended.

The `.gitignore` you are proposing acts like a collection of temporary files more or less related with the python development that are known to the mankind and ignore files like this should be stored in a user-level git config as an optional safe-guard.

Comment (ahmedivy) [reply .gitignore:1]: Its a standard python `.gitignore` file that you can create automatically while creating new Github repo.

[https://github.com/github/gitignore/blob/main/Python.gitignore](https://github.com/github/gitignore/blob/main/Python.gitignore)

If you want, I can change this to include just .venv/ and other compiled files.

**Comment by karpathy** [comment]
I don't love these massive blanked .gitignores.
I think the requirements and the test dupe are good idea.

**Comment by karpathy** [comment]
merged an alternative. took out numpy which is not a dep.

**Pull request closed without merging**